### PR TITLE
OpenWeatherMap: Use api key provider api

### PR DESCRIPTION
**Pull request by mzanetti**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

File: openweathermap/integrationpluginopenweathermap.cpp [hunk 1]
```diff
@@ -96,48 +89,130 @@ void IntegrationPluginOpenweathermap::executeAction(ThingActionInfo *info)
...
 {
-    qCDebug(dcOpenWeatherMap()) << "Refresh data for" << thing->name();
+    qCDebug(dcOpenWeatherMap()) << "Refreshing data for" << thing->name();
     QUrl url("http://api.openweathermap.org/data/2.5/weather");
     QUrlQuery query;
     query.addQueryItem("id", thing->paramValue(openweathermapThingIdParamTypeId).toString());
     query.addQueryItem("mode", "json");
     query.addQueryItem("units", "metric");
-    query.addQueryItem("appid", m_apiKey);
+    query.addQueryItem("appid", apiKeyStorage()->requestKey("openweathermap").data("appid"));
```
Comment (Boernsman) [inline openweathermap/integrationpluginopenweathermap.cpp:107]: please add a plug-in setting for the API key. So the setting's api key would be taken first, if the key is empty the API Key should be used from the API key Storage.

File: openweathermap/integrationpluginopenweathermap.cpp [hunk 2]
```diff
@@ -86,6 +70,15 @@ void IntegrationPluginOpenweathermap::setupThing(ThingSetupInfo *info)
 {
     update(info->thing());
     info->finish(Thing::ThingErrorNoError);
```
Comment (Boernsman) [inline openweathermap/integrationpluginopenweathermap.cpp:72]: Check if any API key is available

**Review by Boernsman — CHANGES_REQUESTED** [review]

File: openweathermap/integrationpluginopenweathermap.cpp (hunk 1 above)
Comment (mzanetti) [reply openweathermap/integrationpluginopenweathermap.cpp]: done

File: openweathermap/integrationpluginopenweathermap.cpp (hunk 2 above)
Comment (mzanetti) [reply openweathermap/integrationpluginopenweathermap.cpp:82]: done

**Review by Boernsman — APPROVED** [review]

**Pull request merged**